src/components/combobox/combobox.tsx

```diff
@@ -386,6 +389,17 @@ export class Combobox implements LabelableComponent, FormComponent, InteractiveC
 
   private activeTransitionProp = "opacity";
 
+  private OpenCloseState: OpenCloseState = {
+    beforeOpen: () => this.calciteComboboxBeforeOpen.emit(this.publicEmitPayload),
+    open: () => this.calciteComboboxOpen.emit(this.publicEmitPayload),
+    beforeClose: () => this.calciteComboboxBeforeClose.emit(this.publicEmitPayload),
+    close: () => this.calciteComboboxClose.emit(this.publicEmitPayload)
+  };
+
+  private publicEmitPayload: any = {
+    element: this
+  };
+
```

Comment:
I split the function up to understand which parts are shared to be extracted and the payload seems to vary (eg. `combobox` only emits `el`, but `alert` also emits a `queue`). 

Are we expecting components to vary on `OpenCloseState` as well (eg. would some components maybe have `close` but not the `open` event emitted)?